cpp/ql/src/semmle/code/cpp/models/implementations/Recv.qll

```diff
@@ -0,0 +1,73 @@
+/**
+ * Provides implementation classes modeling `recv` and various similar
+ * functions. See `semmle.code.cpp.models.Models` for usage information.
+ */
+
+import semmle.code.cpp.models.interfaces.Taint
+import semmle.code.cpp.models.interfaces.ArrayFunction
+import semmle.code.cpp.models.interfaces.Alias
+import semmle.code.cpp.models.interfaces.FlowSource
+import semmle.code.cpp.models.interfaces.SideEffect
+
+/** The function `recv` and its assorted variants */
+private class Recv extends AliasFunction, ArrayFunction, SideEffectFunction, RemoteFlowFunction {
+  Recv() {
+    this.hasGlobalName([
+        "recv", // recv(socket, dest, len, flags)
+        "recvfrom", // recvfrom(socket, dest, len, flags, from, fromlen)
+        "read", // read(socket, dest, len)
+        "pread" // pread(socket, dest, len, offset)
+      ])
```

Comment:
Why should `recvmsg` not be modelled here but only in `Security.qll`?

Reply:
Same reason as for `sendmsg`, really. I chose to not include it in this first PR since it would make the model of the behavior, instead of DefaultTaintTracking, very imprecise. But maybe that's okay for now since that's the behavior we would get from DefaultTaintTracking anyway.

Reply:
It would be imprecise, but I hope it wouldn't be too bad in practice. If we start seeing FPs, we'll have to reconsider.